auction-server/src/state.rs

```diff
@@ -472,16 +495,121 @@ pub struct ChainStoreEvm {
+            &chain_config.geth_rpc_addr,
+            chain_config.rpc_timeout,
+        )
+        .map_err(|err| {
+            tracing::error!(
+                "Failed to create provider for chain({chain_id}) at {rpc_addr}: {:?}",
+                err,
+                chain_id = chain_id,
+                rpc_addr = chain_config.geth_rpc_addr
+            );
+            anyhow!(
+                "Failed to connect to chain({chain_id}) at {rpc_addr}: {:?}",
+                err,
+                chain_id = chain_id,
+                rpc_addr = chain_config.geth_rpc_addr
+            )
+        })?;
+        provider.set_interval(Duration::from_secs(chain_config.poll_interval));
+        Ok(provider)
+    }
+    pub async fn create_store(
+        chain_id: String,
+        config: ConfigEvm,
+        relayer: Wallet<SigningKey>,
+    ) -> anyhow::Result<Self> {
+        let provider = Self::get_chain_provider(&chain_id, &config)?;
+
+        let id = provider.get_chainid().await?.as_u64();
+        let block = provider
+            .get_block(BlockNumber::Latest)
+            .await?
+            .expect("Failed to get latest block");
+
+        let express_relay_contract = get_express_relay_contract(
+            config.express_relay_contract,
+            provider.clone(),
+            relayer.clone(),
+            config.legacy_tx,
+            id,
+        );
+
+
+        Ok(Self {
+            name: chain_id.clone(),
+            core_fields: Default::default(),
+            provider,
+            network_id: id,
+            config: config.clone(),
+            express_relay_contract: Arc::new(express_relay_contract),
+            block_gas_limit: block.gas_limit,
+        })
+    }
+}
+
 pub struct ChainStoreSvm {
     pub core_fields: ChainStoreCoreFields<SimulatedBidSvm>,
 
+    tx_broadcaster_client:             RpcClient,
     pub client:                        RpcClient,
     pub config:                        ConfigSvm,
     pub express_relay_svm:             ExpressRelaySvm,
     pub wallet_program_router_account: Pubkey,
     pub name:                          String,
 }
 
+impl ChainStoreSvm {
+    pub fn new(chain_id: String, config: ConfigSvm, relayer: Arc<Keypair>) -> Self {
+        let express_relay_svm = ExpressRelaySvm {
+            relayer:                     relayer.clone(),
+            permission_account_position: env!("SUBMIT_BID_PERMISSION_ACCOUNT_POSITION")
+                .parse::<usize>()
+                .expect("Failed to parse permission account position"),
+            router_account_position:     env!("SUBMIT_BID_ROUTER_ACCOUNT_POSITION")
+                .parse::<usize>()
+                .expect("Failed to parse router account position"),
+        };
+
+        Self {
+            name: chain_id.clone(),
+            core_fields: Default::default(),
+            client: TracedSenderSvm::new_client(
+                chain_id.clone(),
+                config.rpc_addr.as_str(),
```

Comment:
Maybe we can rename the `rpc_addr` to make the diff clear (with the other rpc). Maybe we can use chatgpt for better naming.